lib/commands/install/InstallTarget.js

```diff
@@ -0,0 +1,258 @@
+var bower = require('bower'),
+	_ = require('lodash'),
+	Q = require('q'),
+	readline = require('readline'),
+	endpointParser = require('bower-endpoint-parser'),
+	semver = require('semver'),
+	Plugin = require('../../Plugin'),
+	InstallLog = require('./InstallLog');
+	extend = require('./extend');
+
+var any = '*';
+var BOWER_MAX_TRY = 5;
+
+var InstallTarget = extend(Plugin, {
+
+	constructor: function(name, requestedVersion, isCompatibleEnabled) {
+        var endpoint = name+'#'+(isCompatibleEnabled ? any : requestedVersion),
+			ep = endpointParser.decompose(endpoint),
+	        version = /^\*$/.test(ep.target) ? any : ep.target,
+	        proposal;
+
+	    Plugin.call(this, ep.name || ep.source, version);
+
+	    if (this.version !== any) {
```

Comment:
tabs AND spaces!

Reply:
sorry I was literally just looking at the big two files in this PR (install and update) and forgot about the dependencies. Is there a preference - tabs or spaces?